### ISSUE TITLE
NullPointerException on players taking damage when the allow-damage flag isn't defined

### ISSUE THREAD

**Issue by ghost** [mewin-archived/WGPlayerDamageFlags#1]
When this plugin is loaded and players take damage, an exception is thrown:

``` java
[SEVERE] Could not pass event EntityDamageByEntityEvent to WGPlayerDamageFlags v0.3
org.bukkit.event.EventException
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:341)
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62)
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:477)
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:462)
        at net.minecraft.server.Explosion.a(Explosion.java:162)
        at net.minecraft.server.World.createExplosion(World.java:1523)
        at net.minecraft.server.WorldServer.createExplosion(WorldServer.java:745)
        at net.minecraft.server.EntityCreeper.h_(EntityCreeper.java:77)
        at net.minecraft.server.World.entityJoinedWorld(World.java:1245)
        at net.minecraft.server.WorldServer.entityJoinedWorld(WorldServer.java:511)
        at net.minecraft.server.World.playerJoinedWorld(World.java:1227)
        at net.minecraft.server.World.tickEntities(World.java:1125)
        at net.minecraft.server.WorldServer.tickEntities(WorldServer.java:428)
        at net.minecraft.server.MinecraftServer.q(MinecraftServer.java:547)
        at net.minecraft.server.DedicatedServer.q(DedicatedServer.java:213)
        at net.minecraft.server.MinecraftServer.p(MinecraftServer.java:474)
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:406)
        at net.minecraft.server.ThreadServerApplication.run(SourceFile:539)
Caused by: java.lang.NullPointerException
        at de.bangl.wgpdf.Utils.dmgAllowedInRegion(Utils.java:120)
        at de.bangl.wgpdf.Utils.dmgAllowedAtLocation(Utils.java:107)
        at de.bangl.wgpdf.listener.PlayerDamageListener.onPlayerDamage(PlayerDamageListener.java:45)
        at sun.reflect.GeneratedMethodAccessor158.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:601)
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:339)
        ... 17 more
```

It's apparently caused by the `allow-damage` flag not being defined for the region (or the lack of a region) where the player takes damage—as if I execute `/region flag region_name allow-damage`, the exception won't be thrown for that region again, even though `/region info` doesn't show any value set for `allow-damage`. It's probably just defining it, but not setting a value.

I've also tried setting `deny-damage`, however an exception is still thrown if that flag is defined but `allow-damage` isn't.

**Comment by BangL** [comment]
thanks for this log! i was unable to reproduce this issue yet. this helps to fix this finally.

**Comment by BangL** [comment]
fixed in v0.4
